src/superphot_plus/model/classifier.py

```diff
@@ -302,6 +171,7 @@ def run(
 
             if val_loss < best_valid_loss:
                 best_valid_loss = val_loss
+                valid_acc = val_acc
                 torch.save(self.state_dict(), model_path)
```

Comment:
I know it is not in your code, but it is likely expensive to save each best model to a file. You might want to consider having a copy of the model in memory and only outputting it at the end.

Reply:
You're right, thanks! Interestingly the impact isn't noticeable, but I decided to apply this change anyway.